### PR TITLE
Fixes #54:Added a ClickableSpan in DrugArrayAdapter Activity

### LINKED ISSUE #54: Added a ClickableSpan in DrugArrayAdapter which takes directly to the DrugSite
What is the use of just displaying the drug sites name user would hardly take the courtesy to copy the link paste it in its search engine and read about it,instead a clickable span would lead it there directly.
Screenshots are below,I have already implemented it can I send a PR?
BEFORE:
screenshot_2016-03-23-21-04-08

AFTER:
screenshot_2016-03-23-23-21-08

### PR DESCRIPTION
**Pull request by HimanshuS01**
Fixes #54 : Added a ClickableSpan in DrugArrayAdapter which takes directly to the DrugSite

**Screen shot for the change**

device-2017-01-26-120149
device-2017-01-26-120031
device-2017-01-26-120122

@chhavip @anubhakushwaha @yatna please review this PR. Thanks :)

### CODE REVIEW

File: malaria-app-android/src/main/java/com/peacecorps/malaria/adapter/DrugArrayAdapter.java [hunk 1]
```diff
@@ -65,12 +69,21 @@ public void addClickListener(final int position, Button infoButton) {
         infoButton.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
-                AlertDialog.Builder dialogAlert  = new AlertDialog.Builder(context);
-                dialogAlert.setMessage(drugDescriptions[position]);
-                dialogAlert.setTitle(dname[position]);
-                dialogAlert.setPositiveButton("OK", null);
-                dialogAlert.setCancelable(true);
-                dialogAlert.create().show();
+
```
Comment (anubhakushwaha) [inline malaria-app-android/src/main/java/com/peacecorps/malaria/adapter/DrugArrayAdapter.java]: Remove extra line

File: malaria-app-android/src/main/java/com/peacecorps/malaria/adapter/DrugArrayAdapter.java [hunk 2]
```diff
@@ -65,12 +69,21 @@ public void addClickListener(final int position, Button infoButton) {
...
+                final SpannableString s = new SpannableString(drugDescriptions[position]);
+                Linkify.addLinks(s, Linkify.ALL);
+
```
Comment (anubhakushwaha) [inline malaria-app-android/src/main/java/com/peacecorps/malaria/adapter/DrugArrayAdapter.java]: Same

File: malaria-app-android/src/main/java/com/peacecorps/malaria/adapter/DrugArrayAdapter.java [hunk 3]
```diff
@@ -65,12 +69,21 @@ public void addClickListener(final int position, Button infoButton) {
...
+                final AlertDialog d = new AlertDialog.Builder(getContext())
+                        .setPositiveButton("OK", null)
+                        .setMessage( s )
+                        .setTitle(dname[position])
+                        .create();
+
+                d.show();
+
```
Comment (anubhakushwaha) [inline malaria-app-android/src/main/java/com/peacecorps/malaria/adapter/DrugArrayAdapter.java]: Same

File: malaria-app-android/src/main/java/com/peacecorps/malaria/adapter/DrugArrayAdapter.java [hunk 4]
```diff
@@ -65,12 +69,21 @@ public void addClickListener(final int position, Button infoButton) {
...
+                // Make the textview clickable. Must be called after show()
+                ((TextView)d.findViewById(android.R.id.message)).setMovementMethod(LinkMovementMethod.getInstance());
+
```
Comment (anubhakushwaha) [inline malaria-app-android/src/main/java/com/peacecorps/malaria/adapter/DrugArrayAdapter.java]: Same

**Comment by HimanshuS01** [comment]
Extra lines has been removed.
@anubhakushwaha @yatna @chhavip Please review. Thanks

**Comment by HimanshuS01** [comment]
Thank you for approving this PR.
@anubhakushwaha Please also review  the  PR #219

**Comment by free4murad** [comment]
@HimanshuS01 
You might wanna change the base branch.

**Comment by HimanshuS01** [comment]
@free4murad As my previous PR's has been merged into the master branch only that is why I choose that branch.
@anubhakushwaha should I need to change base branch or not? Thanks.

**Comment by anubhakushwaha** [comment]
@HimanshuS01 Yes please,change the base branch to develop

**Comment by HimanshuS01** [comment]
@anubhakushwaha when I changed my base branch to develop,All the previous commits(around 97 commits) in develop branch are showing  in my PR along with my commit.
There is some problem arising when I update my origin/develop branch with the upstream/develop.
I will clone the project again and send the PR's to develop branch for my another issue from the next time.Please review this time. Thanks :)

**Comment by HimanshuS01** [comment]
@anubhakushwaha @chhavip @yatna I have changed the base branch. Please review and merge this PR. Thanks :)

File: malaria-app-android/src/main/java/com/peacecorps/malaria/adapter/DrugArrayAdapter.java [hunk 5]
```diff
@@ -65,12 +69,16 @@ public void addClickListener(final int position, Button infoButton) {
...
             public void onClick(View v) {
-                AlertDialog.Builder dialogAlert  = new AlertDialog.Builder(context);
-                dialogAlert.setMessage(drugDescriptions[position]);
-                dialogAlert.setTitle(dname[position]);
-                dialogAlert.setPositiveButton("OK", null);
-                dialogAlert.setCancelable(true);
-                dialogAlert.create().show();
+                final SpannableString s = new SpannableString(drugDescriptions[position]);
+                Linkify.addLinks(s, Linkify.ALL);
+                final AlertDialog d = new AlertDialog.Builder(getContext())
```
Comment (chhavip) [inline malaria-app-android/src/main/java/com/peacecorps/malaria/adapter/DrugArrayAdapter.java]: Use a better local variable name.

File: malaria-app-android/src/main/java/com/peacecorps/malaria/adapter/DrugArrayAdapter.java [hunk 6]
```diff
@@ -65,12 +69,16 @@ public void addClickListener(final int position, Button infoButton) {
...
+                        .setPositiveButton("OK", null)
+                        .setMessage( s )
```
Comment (chhavip) [inline malaria-app-android/src/main/java/com/peacecorps/malaria/adapter/DrugArrayAdapter.java]: No need for extra spaces

File: malaria-app-android/src/main/java/com/peacecorps/malaria/adapter/DrugArrayAdapter.java (hunk 5 above)
Comment (chhavip) [inline malaria-app-android/src/main/java/com/peacecorps/malaria/adapter/DrugArrayAdapter.java]: Specify the reason for declaring the variables to be final or declare them as normal variables if not needed.

Comment (HimanshuS01) [inline malaria-app-android/src/main/java/com/peacecorps/malaria/adapter/DrugArrayAdapter.java]: sure

File: malaria-app-android/src/main/java/com/peacecorps/malaria/adapter/DrugArrayAdapter.java (hunk 6 above)
Comment (HimanshuS01) [inline malaria-app-android/src/main/java/com/peacecorps/malaria/adapter/DrugArrayAdapter.java]: ok

File: malaria-app-android/src/main/java/com/peacecorps/malaria/adapter/DrugArrayAdapter.java (hunk 5 above)
Comment (HimanshuS01) [inline malaria-app-android/src/main/java/com/peacecorps/malaria/adapter/DrugArrayAdapter.java]: sure.

**Comment by HimanshuS01** [comment]
@chhavip Thanks for reviewing this PR. I have done the requested changes.

**Pull request merged**